Our Userland Makefile replaces the upstream link-editor version script with a
Solaris-specific version 2 mapfile which is incompatible with the
ld --version-script option. Remove the upstream configure test that prefers
to use --version-script if available, and force the use of -M.

The patch is not suitable for upstream.

--- libxslt-1.1.37/configure.ac	2022-08-29 16:29:34.000000000 +0000
+++ libxslt-1.1.37/configure.ac.new	2022-12-26 17:27:31.800211970 +0000
@@ -93,12 +93,7 @@
 dnl then add it
 dnl
 AC_MSG_CHECKING([how to pass version script to the linker ($LD)])
-VERSION_SCRIPT_FLAGS=none
-if $LD --help 2>&1 | grep "version-script" >/dev/null 2>/dev/null; then
-    VERSION_SCRIPT_FLAGS=-Wl,--version-script=
-elif $LD --help 2>&1 | grep "M mapfile" >/dev/null 2>/dev/null; then
-    VERSION_SCRIPT_FLAGS="-Wl,-M -Wl,"
-fi
+VERSION_SCRIPT_FLAGS="-Wl,-M -Wl,"
 AC_MSG_RESULT([$VERSION_SCRIPT_FLAGS])
 AC_SUBST(VERSION_SCRIPT_FLAGS)
 AM_CONDITIONAL([USE_VERSION_SCRIPT], [test "$VERSION_SCRIPT_FLAGS" != none])
